app/global/MainModule.java

```diff
@@ -186,6 +197,42 @@ private ObjectMapper provideObjectMapper(final ApplicationLifecycle lifecycle) {
         return objectMapper;
     }
 
+    @Provides
+    @Singleton
+    @Named("job-execution-shard-region")
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
+    private ActorRef provideJobExecutorShardRegion(
+            final ActorSystem system,
+            final Injector injector,
+            final JobMessageExtractor extractor,
+            final Clock clock,
+            final PeriodicMetrics periodicMetrics) {
+        final ClusterSharding clusterSharding = ClusterSharding.get(system);
+        return clusterSharding.start(
+                "JobExecutor",
```

Comment:
So for the job framework do we see having a single cluster of job executors for all types of jobs (e.g. reports, alerts, etc.) or do we see creating separate "rings" of these actors? If it's the former, we probably need to increase the number of partitions a fair bit. If it's the latter, I suspect we need to parameterize this code some more.

Reply:
"Rings"? Are you suggesting having one `ClusterSharding` instance per Organization?

I think I favor the "single ClusterSharding, single anti-entropy actor" model: the fact that messages to `JobExecutorActor`s are routed based on Organization makes me feel safe that different organizations' actors will be well-isolated from each other.

(This could change when there are so many jobs on the cluster that the AE / Region routing actor can't keep up; but at this point that seems like a distant dream, and that layer of indirection seems as easy to add later as to add now.)

Reply:
Ah, no, you were suggesting having one `ClusterSharding` per... `JobRepository` instance?